lib/ssl/src/ssl_session.erl

```diff
@@ -66,17 +67,19 @@ is_new(_ClientSuggestion, _ServerDecision) ->
 %%              for the client hello message.
 %%--------------------------------------------------------------------
 client_select_session({_, _, #{versions := Versions,
-                               protocol := Protocol}} = ClientInfo, 
+                               protocol := Protocol} = Opts} = ClientInfo,
                       Cache, CacheCb, NewSession, CertKeyPairs) ->
     
     RecordCb = record_cb(Protocol),
-    Version = RecordCb:lowest_protocol_version(Versions),
+    LVersion = RecordCb:lowest_protocol_version(Versions),
+    HVersion = RecordCb:highest_protocol_version(Versions),
     
-    case Version of
+    case LVersion of
```

Comment:
I think we need comment in the code here which explain what the case statement and the logic is supposed to do.
Maybe something like:
If the lowest acceptable protocol version is TLS 1.3 ({3,N} when N>=4 then .....
else if the lowest acceptable protocol version is TLS 1.2 or even lower then ... and if highest acceptable protocol version is TLS 1.3 then we do what?

Reply:
Added comment!